File: automobileinsuranceapi/src/main/java/com/hexaware/automobileinsurance/service/ClaimService.java
package com.hexaware.automobileinsurance.service;

import java.util.List;

import com.hexaware.automobileinsurance.dto.ClaimDTO;
import com.hexaware.automobileinsurance.model.Claim;

public interface ClaimService {

    Claim raiseClaim(ClaimDTO dto);

    Claim getClaimById(Integer claimId);

    List<Claim> getAllClaims();

    Claim updateClaimStatus(Integer claimId, ClaimDTO dto);
    
    Claim approveClaim(Integer claimId);

    Claim rejectClaim(Integer claimId);
}

File: automobile-insurance-frontend/src/services/ClaimService.js
import api from "../api/api";

const ClaimService = {

    getAllClaims() {

        return api.get("/claims");

    },

    getClaimById(id) {

        return api.get(`/claims/${id}`);

    },

    raiseClaim(claim) {

        return api.post("/claims", claim);

    },

    updateClaim(id, claim) {

        return api.put(`/claims/${id}`, claim);

    },

    approveClaim(id) {

        return api.put(`/claims/${id}/approve`);

    },

    rejectClaim(id) {

        return api.put(`/claims/${id}/reject`);

    }

};

export default ClaimService;